retriever/lib/scripts.py

```diff
@@ -107,7 +108,101 @@ def get_script(dataset):
     if dataset in scripts:
         return scripts[dataset]
     else:
-        raise KeyError("No dataset named: {}".format(dataset))
+        if dataset in RETRIEVER_DATASETS:
+            read_script = get_script_upstream(dataset, repo=RETRIEVER_REPOSITORY)
+        else:
+            read_script = get_script_upstream(dataset)
+        if read_script is None:
+            raise KeyError("No dataset named: {}".format(dataset))
+        else:
+            return read_script
+
+
+def get_script_upstream(dataset, repo=REPOSITORY):
+    """Return the upstream script for a named dataset."""
+    try:
+        is_json = True
+        script = dataset.replace('-', '_')
+        script_name = script + ".json"
+        filepath = "scripts/" + script_name
+        newpath = os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name))
+        r = requests.get(repo + filepath, allow_redirects=True, stream=True)
+        if r.status_code == 404:
+            is_json = False
+            script_name = script + ".py"
+            filepath = "scripts/" + script_name
+            newpath = os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name))
+            r = requests.get(repo + filepath, allow_redirects=True, stream=True)
+            if r.status_code == 404:
+                return None
+        with open(newpath, 'wb') as f:
+            for chunk in r.iter_content(chunk_size=1024):
+                f.write(chunk)
+        r.close()
+        if is_json:
+            read_script = read_json(join(SCRIPT_WRITE_PATH, script))
+            setattr(read_script, "_file", os.path.join(SCRIPT_WRITE_PATH, script_name))
+            setattr(read_script, "_name", script)
+            return read_script
+        else:
+            file, pathname, desc = imp.find_module(script, [SCRIPT_WRITE_PATH])
+            new_module = imp.load_module(script, file, pathname, desc)
+            setattr(new_module.SCRIPT, "_file", os.path.join(SCRIPT_WRITE_PATH, script_name))
+            setattr(new_module.SCRIPT, "_name", script)
+            return new_module.SCRIPT
+    except:
+        raise
```

Comment:
Downloads the script from the upstream repo (mentioned in the argument), and then returns it as an instance of `BasicTextTemplate` taking into consideration whether it is a json or a python script.